### PR TITLE
Support for adding/removing metadata values in the emulator.

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py [hunk 1]
```diff
@@ -664,6 +770,22 @@ def __init__(self, items, lock):
...
     MetadataDirectory.__init__(self, items, lock)
 
+  def deleteChild(self, child_name):
+    """Attempts to delete the child with the specified name and returns whether the operation was successful.
+
+    Args:
+      child_name: The name where the value should be deleted.
+    """
+    try:
+      index = int(child_name)
```
Comment (aleaxit) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py]: just on principle, keep the body of the try clause as tight as feasible: here, just insert an `else:` after line 780

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py [hunk 2]
```diff
@@ -529,6 +596,27 @@ def childIter(self):
...
+
+    Args:
+      child_name: The name where the new value will be placed.
+      lock: the lock to use to create new metadata values if necessary.
+    """
+    if self.hasFixedKeys():
+      return self.getChild(child_name)
+
+    for child in self.childIter():
+      return child
```
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py]: Done

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py (hunk 1 above)
Comment (aleaxit) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py]: Your code is fine (the else: clause goes after the except: clause, just as you've placed it) -- my suggestion was imprecise (it implied placing the else: before the except:, which would be invalid syntax).

Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py]: Ok, thank you.

**Pull request merged**